src/frequenz/sdk/microgrid/component/_component.py

```diff
@@ -8,9 +8,42 @@
 from dataclasses import dataclass
 from enum import Enum
 
+import frequenz.api.microgrid.inverter_pb2 as inverter_pb
 import frequenz.api.microgrid.microgrid_pb2 as microgrid_pb
 
 
+class ComponentType(Enum):
+    """A ABC from which individual component types are derived."""
+
+
+class InverterType(ComponentType):
+    """Enum representing inverter types."""
+
+    NONE = inverter_pb.Type.TYPE_UNSPECIFIED
+    BATTERY = inverter_pb.Type.TYPE_BATTERY
+    SOLAR = inverter_pb.Type.TYPE_SOLAR
+    HYBRID = inverter_pb.Type.TYPE_HYBRID
+
+
+def _component_type_from_protobuf(
+    component_type: inverter_pb.Type.ValueType,
+) -> ComponentType:
```

Comment:
Am I crazy or the user of this function will have to forcibly cast the return to `InverterType` to be of any use?

Reply:
A typehint from a super class doesn't change the type of the object being returned, in Python.  So the function still returns `InverterType.BATTERY` values that's directly comparable.  You're forgetting that we're using Python,  so yes, you are crazy \o/

Reply:
Ah, no, I remember this is Python, but the type checker is still a type checker and it won't let you do this:

```py
ct = _component_type_from_protobuf(...)
if ct == ct.BATTERY:
    ...
```

for example, but of course this is not what you want, the trick is you are still always comparing with the specialized type member:

```py
ct = _component_type_from_protobuf(...)
if ct == InverterType.BATTERY:
    ...
```

So you don't need to access any members in the return type, just compare it.

So yeah, all good, sorry about the brain fart :dash: 🤭